x-pack/plugins/security_solution/server/endpoint/routes/actions/response_actions.test.ts

```diff
@@ -0,0 +1,555 @@
+import { licenseMock } from '@kbn/licensing-plugin/common/licensing.mock';
+import { License } from '@kbn/licensing-plugin/common/license';
+import { AwaitedProperties } from '@kbn/utility-types';
+import {
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RouteConfig,
+} from '@kbn/core/server';
+import {
+  elasticsearchServiceMock,
+  httpServerMock,
+  httpServiceMock,
+  loggingSystemMock,
+  savedObjectsClientMock,
+} from '@kbn/core/server/mocks';
+import type { PackageClient } from '@kbn/fleet-plugin/server';
+import { createMockPackageService } from '@kbn/fleet-plugin/server/mocks';
+import { AGENT_ACTIONS_INDEX, ElasticsearchAssetType } from '@kbn/fleet-plugin/common';
+import { CasesClientMock } from '@kbn/cases-plugin/server/client/mocks';
+
+import { parseExperimentalConfigValue } from '../../../../common/experimental_features';
+import { LicenseService } from '../../../../common/license';
+import {
+  ISOLATE_HOST_ROUTE_V2,
+  RELEASE_HOST_ROUTE,
+  metadataTransformPrefix,
+  ENDPOINT_ACTIONS_INDEX,
+} from '../../../../common/endpoint/constants';
+import {
+  ActionDetails,
+  EndpointAction,
+  HostIsolationRequestBody,
+  ResponseActionResponse,
+  HostMetadata,
+  LogsEndpointAction,
+} from '../../../../common/endpoint/types';
+import { EndpointDocGenerator } from '../../../../common/endpoint/generate_data';
+import { EndpointAuthz } from '../../../../common/endpoint/types/authz';
+import { createMockConfig } from '../../../lib/detection_engine/routes/__mocks__';
+import { SecuritySolutionRequestHandlerContextMock } from '../../../lib/detection_engine/routes/__mocks__/request_context';
+import { EndpointAppContextService } from '../../endpoint_app_context_services';
+import {
+  createMockEndpointAppContextServiceSetupContract,
+  createMockEndpointAppContextServiceStartContract,
+  createRouteHandlerContext,
+} from '../../mocks';
+import { legacyMetadataSearchResponseMock } from '../metadata/support/test_support';
+import { registerResponseActionRoutes } from './response_actions';
+import * as ActionDetailsService from '../../services/actions/action_details_by_id';
+
+// jest.mock('../../services/actions/action_details_by_id', () => ({ getActionDetailsById: jest.fn() }));
+
+interface CallRouteInterface {
+  body?: HostIsolationRequestBody;
+  idxResponse?: any;
+  searchResponse?: HostMetadata;
+  mockUser?: any;
+  license?: License;
+  authz?: Partial<EndpointAuthz>;
+}
+
+const Platinum = licenseMock.createLicense({ license: { type: 'platinum', mode: 'platinum' } });
+const Gold = licenseMock.createLicense({ license: { type: 'gold', mode: 'gold' } });
+
+describe('Response actions', () => {
+  describe('handler', () => {
+    let endpointAppContextService: EndpointAppContextService;
+    let mockResponse: jest.Mocked<KibanaResponseFactory>;
+    let licenseService: LicenseService;
+    let licenseEmitter: Subject<ILicense>;
+    let getActionDetailsByIdSpy: jest.SpyInstance;
```

Comment:
this spy is new, added to spy/mock getActionDetailsById for the newly included ActionDetails in response.